### PR TITLE
fix(json): dictionary encoding

### LINKED ISSUE #3603: JSON encoding fails for dictionary type
Dictionary seems like a great type for optional "schema-less" (unlike records) arguments for interacting with external systems. JSON encoding is common, however in Flux `json.encode` fails for with `Error: error calling function "encode" @1:1-1:27: json: unsupported type: map[interface {}]interface {}`. 

Example:
```
import "json"
attributes = ["region": "EU"]
json.encode(v: attributes)
```

Since Go maps are encoded as objects, and objects keys are strings in JSON, wouldn't it make sense to force keys into strings and use `map[string]interface{}` just like for Flux record type?

### PR DESCRIPTION
**Pull request by alespour**
This PR fixes #3603. 

Since Go maps are encoded as JSON objects where keys are strings, map key is changed from `interface{}` to `string` and keys values are simply formated as strings.

Example:
```
import "json"
attributes = ["region": "EU"]
json.encode(v: attributes)
```

Returns:
```
{"region": "EU"}
```

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

**Comment by alespour** [comment]
@jsternberg Do you think this makes sense? Thank you.

File: stdlib/json/encode.go [hunk 1]
```diff
@@ -100,7 +101,7 @@ func convertValue(v values.Value) (interface{}, error) {
 		return nil, errors.New(codes.Invalid, "cannot encode a function value")
 	case semantic.Dictionary:
 		dict := v.Dict()
-		d := make(map[interface{}]interface{}, dict.Len())
+		d := make(map[string]interface{}, dict.Len())
```
Comment (jsternberg) [inline stdlib/json/encode.go:104]: Can you make a comment about how json only allows string keys? I didn't know that before reading this: https://www.json.org/json-en.html.

**Review by jsternberg — CHANGES_REQUESTED** [review]
Can you add a comment to the spec in the docs folder describing this behavior? It seems fine I just want it to get explicitly documented.

**Comment by alespour** [comment]
> Can you add a comment to the spec in the docs folder describing this behavior? It seems fine I just want it to get explicitly documented.

I'm sorry, do you mean somewhere to SPEC.md or Outputs.md? I'm not sure where such comment fits better.

**Comment by jsternberg** [comment]
I mean `docs/SPEC.md`. If this function has an entry there, that should be updated to reflect this change.

**Comment by alespour** [comment]
@jsternberg sorry, this PR fell into oblivion for some time, but now it popped up again in similar context as before. The code updated to current master, `docs/SPEC.md` also updated.

**Comment by scbrickley** [comment]
Closed in favor of https://github.com/influxdata/flux/pull/4550

**Pull request closed without merging**